### PR TITLE
added logic to further enable the fetching of meta

### PR DESCRIPTION
**Pull request by nicktaras**

### CODE REVIEW

**Comment by nicktaras** [comment]
From testing, added some additional ways to return the key API data for Solana.

File: src/services/solanaTokenDiscoveryService.ts [hunk 1]
```diff
@@ -480,8 +480,8 @@ export default class SolanaTokenDiscoveryService {
         return response
           .map((item: any) => {
 
-            const tokenId = this.getIdFromString(item?.metaData?.name);
-            const name = item?.metaData?.name || item?.tokenInfo?.name;
+            const tokenId = this.getIdFromString(item?.metaData?.name || item?.data?.name);
```
Comment (micwallace) [inline src/services/solanaTokenDiscoveryService.ts:483]: @nicktaras one change I'd suggest is that tokenId is not derived from the metadata name. 
We can have this as a separate field if nessesary. Unique tokenId may be required later for authentication. 
EVM implementation currently returns actual token ID, Solana should be consistent with this.

**Review by micwallace — CHANGES_REQUESTED** [review]

**Review by jeanhwa — APPROVED** [review]
Tested and working

File: src/services/solanaTokenDiscoveryService.ts (hunk 1 above)
Comment (micwallace) [reply src/services/solanaTokenDiscoveryService.ts:483]: The actual token ID is available in both Solscan & Moralis. The field is named "mint" in Moralis & "tokenAddress" on Solscan. This is the unique token ID which is the equivalent to contractAddress+tokenId on an EVM chain. 

So I propose using this as tokenId, and if required we can have "tokenIdName" or some other field to have the collection number derived from the metadata name.

**Pull request merged**